testbed/testbed/test_case.go

```diff
@@ -169,7 +169,9 @@ func (tc *TestCase) StartAgent(args ...string) {
 	}()
 
 	endpoint := tc.LoadGenerator.sender.GetEndpoint()
-	if endpoint != nil {
+
+	// Prometheus endpoints use /metrics path instead of only tcp host and port
+	if endpoint != nil && tc.LoadGenerator.sender.ProtocolName() != "prometheus" {
```

Comment:
We should not be doing exceptions like this in the generic code. Please find a different way.

Reply:
This is tough since for the prometheus sender,`tc.LoadGenerator.sender.GetEndpoint()` is non-nil and is used to know what endpoint the sender is using to host the metrics, which is needed in the prometheus config of the prometheus receiver. However, the prometheus sender actually uses `<host>:<port>/metrics`, so the function `net.Dial(tc.LoadGenerator.sender.GetEndpoint().Network(), tc.LoadGenerator.sender.GetEndpoint().String())` errors since just the `<host>:<port>` endpoint isn't setup, and because of this error, the whole load test fails.

I have tried changing to have `endpointString` as a property of `PrometheusDataSender`, and not use the port and host of `DataSenderBase` but `GetEndpoint()` still returns `:0` since the `port` of type int defaults to 0. Should `GetEndpoint()` instead return nil if `Host` is an empty string and `Port` is 0?